tests/generic_relations_regress/tests.py

```diff
@@ -140,14 +139,37 @@ def count_places(place):
         self.assertEqual(count_places(p1), 1)
         self.assertEqual(count_places(p2), 1)
 
-    def test_target_model_is_unsaved(self):
-        """Test related to #13085"""
-        # Fails with another, ORM-level error
+    def test_target_model_is_unsaved_save(self):
+        # Assigning an unsaved object to a GFK should raise a ValueError on save.
         dev1 = Developer(name="Joe")
         note = Note(note="Deserves promotion", content_object=dev1)
-        with self.assertRaises(IntegrityError):
+        msg = (
+            "save() prohibited to prevent data loss due to unsaved related object "
+            "'content_object'."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
             note.save()
 
+    def test_target_model_is_unsaved_bulk_create(self):
+        # Assigning an unsaved object to a GFK should raise a ValueError on bulk create.
+        dev1 = Developer(name="Joe")
+        note = Note(note="Deserves promotion", content_object=dev1)
+        msg = (
+            "bulk_create() prohibited to prevent data loss due to unsaved related "
+            "object 'content_object'."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            Note.objects.bulk_create([note])
+
+    def test_save_target_model_is_unsaved(self):
+        dev1 = Developer(name="Joe")
+        note = Note(note="Deserves promotion", content_object=dev1)
+        dev1.save()
+        # ValueError is not raised as unsaved object since been saved.
+        note.save()
+        note.refresh_from_db()
+        self.assertEqual(note.content_object, dev1)
```

Comment:
It looks like a duplicate of `test_save_unsaved_instance_on_generic_foreign_key` :thinking:

Reply:
Removed - I had the same feeling with the other two tests above (`test_target_model_is_unsaved_save` vs `test_unsaved_instance_on_generic_foreign_key_save` and `test_target_model_is_unsaved_bulk_create` vs `test_unsaved_instance_on_generic_foreign_key_bulk_create`) but as two tests existed before, I wasn't sure if I had missed something
Shall I also remove `test_target_model_is_unsaved_save` and `test_target_model_is_unsaved_bulk_create`?

Reply:
> Shall I also remove `test_target_model_is_unsaved_save` and `test_target_model_is_unsaved_bulk_create`?

Yes, they are redundant.

Reply:
They're dead :+1: